### PR TITLE
Migrate Discord module to JDA 5

### PR DESCRIPTION
**Pull request by JRoy**
Updates EssentialsXDiscord to use the v5 release of JDA

No change in API and link module will require zero updates, as both don't touch JDA at all.

merge https://github.com/EssentialsX/Website/pull/92 with this

REBASE AND MERGE

### CODE REVIEW

**Review by pop4959 — COMMENTED** [review]
Looks good so far

**Comment by JRoy** [comment]
TODO: use jda's `SplitUtil` in next JDA bump

**Comment by JRoy** [comment]
> TODO: use jda's `SplitUtil` in next JDA bump

nvm my code is better

File: EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ public void startup() throws LoginException, InterruptedException {
 
         jda = JDABuilder.createDefault(plugin.getSettings().getBotToken())
                 .addEventListeners(new DiscordListener(this))
-                .enableCache(CacheFlag.EMOTE)
+                .enableCache(CacheFlag.EMOJI)
```
Comment (Andre601) [inline EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java:173]: Is that even needed?
createDefault does to my knowledge cache emojis by default, so this enable call is somewhat useless

File: EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java [hunk 2]
```diff
@@ -192,7 +192,7 @@ public void startup() throws LoginException, InterruptedException {
         }
 
         // Load emotes into cache, JDA will handle updates from here on out.
-        guild.retrieveEmotes().queue();
+        guild.retrieveEmojis().queue();
```
Comment (Andre601) [inline EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java:209]: Refer to my previous point. JDA usually does cache emojis when it gets the Guild into the cache.

File: EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java (hunk 1 above)
Comment (JRoy) [reply EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java:173]: Want this explicit incase of any change in behavior that I miss during an update

File: EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java (hunk 2 above)
Comment (JRoy) [reply EssentialsDiscord/src/main/java/net/essentialsx/discord/JDADiscordService.java:209]: ditto

**Review by mdcfe — APPROVED** [review]

**Pull request merged**